src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java

```diff
@@ -20,27 +20,49 @@
  * @author Alec Minchington
  *
  */
-public class ProfileHandler implements Cloneable {
+public class ProfileHandler {
 
 	/**
-	 * The motion profile to execute.
+	 * The motion profile to execute on {@link #leftTalon}.
 	 */
-	private double[][] profile;
-	
+	private double[][] leftProfile;
+
 	/**
-	 * Represents the current point being streamed from the profile.
+	 * The motion profile to execute on {@link #rightTalon}.
 	 */
-	int profileIndex = 0;
+	private double[][] rightProfile;
 
 	/**
-	 * The PID gains profile {@link #talon} will use to execute the motion profile.s
+	 * Represents the current point being streamed from the left profile to the left talon.
 	 */
-	private int gainsProfile;
+	int leftProfileIndex = 0;
 
 	/**
-	 * The Talon used to execute the motion profile.
+	 * Represents the current point being streamed from the right profile to the right talon.
 	 */
-	private TalonSRX talon;
+	int rightProfileIndex = 0;
```

Comment:
Pairing profiles together only makes sense if they're the same length, I don't think you need separate indexes.